imageio/plugins/pillow_legacy.py

```diff
@@ -213,6 +214,12 @@
         mode 'F'.
 """
 
+warnings.warn(
+    "The `PillowFormat` plugin is deprecated and will be removed in ImageIO v3."
+    " Use the new `PillowPlugin` instead.",
+    DeprecationWarning,
+)
```

Comment:
When does a user get this warning? Most users won't know how to select the `PillowPlugin` over the `PillowFormat`, so I'm not sure how useful this warning is ...

Reply:
For users that sidestep our normal plugin selection mechanism and that access the format directly, e.g., via `iio.formats["PNG-PIL"]`. I expect this to be niche, but it is better to warn than to just drop it :)

At the moment this will also warn for some of our codepaths, but this is a thing that I have to change (and one of the reasons this is still a draft).